app/src/pages/inside/settingsPage/settingsTabs/notificationsTab/forms/notificationRuleForm/validation.js

```diff
@@ -0,0 +1,47 @@
+const compareRules = (ruleOne, ruleTwo) => {
+  if (ruleOne.launchStatsRule !== ruleTwo.launchStatsRule) {
+    return false;
+  }
+  if (ruleOne.launchTagRule.length !== ruleTwo.launchTagRule.length) {
+    return false;
+  }
+  if (ruleOne.launchNameRule.length !== ruleTwo.launchNameRule.length) {
+    return false;
+  }
+  if (ruleOne.recipients.length !== ruleTwo.recipients.length) {
+    return false;
+  }
+  return (
+    ruleOne.launchTagRule.sort().every((tag) => ruleTwo.launchTagRule.includes(tag)) &&
+    ruleOne.launchNameRule.sort().every((launch) => ruleTwo.launchNameRule.includes(launch)) &&
+    ruleOne.recipients.sort().every((recipient) => ruleTwo.recipients.includes(recipient))
+  );
+};
+
+export const validate = ({ rules }) => {
+  const errors = {};
+  const rulesArrayErrors = [];
+
+  rules.forEach((item, index, arr) => {
```

Comment:
This function is difficult to understand (see comments below). I'm not sure, but will my version be correct too?
```
const rulesErrors = rules.map((rule, index) => {
  if (rules.length > 1) {
    const hasDuplicates = arr.find((item, i) => compareRules(item, rule) && i !== index);
    if (hasDuplicates) {
      return { confirmed: 'hasDuplicates' };
    }
  }
  if (rule.recipients.length === 0 && !rule.informOwner) {
    return {
      recipients: 'recipientsHint',
      confirmed: 'error',
    };
  }
  return null;
}).filter(Boolean);
```

Reply:
But these errors in different if blocks can be found at the same time. So, I use your solution with a few updates.

Reply:
I just change the order of these conditions according to their priority